storage/src/adb/any/mod.rs

```diff
@@ -2,472 +2,169 @@
-        } else {
-            let steps_to_take = self.steps + 1;
-            for _ in 0..steps_to_take {
-                let loc = self.inactivity_floor_loc;
-                self.inactivity_floor_loc = self
-                    .as_floor_helper()
-                    .raise_floor_with_bitmap(status, loc)
-                    .await?;
-            }
-        }
-        self.steps = 0;
-
-        Ok(self.inactivity_floor_loc)
-    }
-
-    /// Returns a FloorHelper wrapping the current state of the log.
-    pub(super) fn as_floor_helper(
-        &mut self,
-    ) -> FloorHelperState<'_, E, C, O, I, H, T, Clean<DigestOf<H>>> {
-        FloorHelper {
-            snapshot: &mut self.snapshot,
-            log: &mut self.log,
-            translator: PhantomData,
-        }
-    }
+    ) -> impl Future<Output = Result<(Proof<H::Digest>, Vec<O>), Error>>;
+}
+
+/// Configuration for an `Any` authenticated db with fixed-size values.
+#[derive(Clone)]
+pub struct FixedConfig<T: Translator> {
+    /// The name of the [Storage] partition used for the MMR's backing journal.
+    pub mmr_journal_partition: String,
+
+    /// The items per blob configuration value used by the MMR journal.
+    pub mmr_items_per_blob: NonZeroU64,
+
+    /// The size of the write buffer to use for each blob in the MMR journal.
+    pub mmr_write_buffer: NonZeroUsize,
+
+    /// The name of the [Storage] partition used for the MMR's metadata.
+    pub mmr_metadata_partition: String,
+
+    /// The name of the [Storage] partition used to persist the (pruned) log of operations.
+    pub log_journal_partition: String,
+
+    /// The items per blob configuration value used by the log journal.
+    pub log_items_per_blob: NonZeroU64,
+
+    /// The size of the write buffer to use for each blob in the log journal.
+    pub log_write_buffer: NonZeroUsize,
+
+    /// The translator used by the compressed index.
+    pub translator: T,
+
+    /// An optional thread pool to use for parallelizing batch operations.
+    pub thread_pool: Option<ThreadPool>,
+
+    /// The buffer pool to use for caching data.
+    pub buffer_pool: PoolRef,
+}
+
+/// Configuration for an `Any` authenticated db with variable-sized values.
+#[derive(Clone)]
+pub struct VariableConfig<T: Translator, C> {
+    /// The name of the [Storage] partition used for the MMR's backing journal.
+    pub mmr_journal_partition: String,
+
+    /// The items per blob configuration value used by the MMR journal.
+    pub mmr_items_per_blob: NonZeroU64,
+
+    /// The size of the write buffer to use for each blob in the MMR journal.
+    pub mmr_write_buffer: NonZeroUsize,
+
+    /// The name of the [Storage] partition used for the MMR's metadata.
+    pub mmr_metadata_partition: String,
+
+    /// The name of the [Storage] partition used to persist the log of operations.
+    pub log_partition: String,
+    /// The size of the write buffer to use for each blob in the log journal.
```

Comment:
nit
```suggestion

    /// The size of the write buffer to use for each blob in the log journal.
```

Reply:
Also, do we want to align naming between `FixedConfig` and `VariableConfig` now that they're adjacent? e.g. `log_items_per_section` vs `log_items_per_blob`. I'd be OK deferring this to keep the PR smaller.

Reply:
done & done